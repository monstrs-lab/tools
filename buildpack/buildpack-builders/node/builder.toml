[[buildpacks]]
id = "tech.monstrs.buildpacks.yarn-install"
version = "0.0.1"
image = "monstrs/buildpack-yarn-install:0.0.1"

[[buildpacks]]
id = "tech.monstrs.buildpacks.node-start"
version = "0.0.1"
image = "monstrs/buildpack-node-start:0.0.1"

[[buildpacks]]
id = "tech.monstrs.buildpacks.yarn-workspace-start"
version = "0.0.1"
image = "monstrs/buildpack-yarn-workspace-start:0.0.1"

[[order]]
[[order.group]]
id = "tech.monstrs.buildpacks.yarn-install"
version = "0.0.1"

[[order.group]]
id = "tech.monstrs.buildpacks.node-start"
version = "0.0.1"

[[order.group]]
id = "tech.monstrs.buildpacks.yarn-workspace-start"
version = "0.0.1"

[stack]
id = "tech.monstrs.stacks.node"
run-image = "monstrs/stack-node:run"
build-image = "monstrs/stack-node:build"
