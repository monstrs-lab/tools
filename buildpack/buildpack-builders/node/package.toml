[[dependencies]]
  image = "monstrs/buildpack-yarn-install:0.0.1"

[[dependencies]]
  image = "monstrs/buildpack-node-start:0.0.1"

[[dependencies]]
  image = "monstrs/buildpack-yarn-workspace-start:0.0.1"
