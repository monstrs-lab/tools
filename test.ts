const n = (a: string) => a

n(5)
