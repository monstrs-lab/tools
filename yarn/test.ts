const n = (a: string) => a
console.log(n(5))
