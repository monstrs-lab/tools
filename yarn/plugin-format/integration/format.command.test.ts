import assert         from 'node:assert/strict'
import { beforeEach } from 'node:test'
import { test }       from 'node:test'

import { TestEnv }    from '@monstrs/yarn-test-utils'

let testEnv: TestEnv

beforeEach(async () => {
  testEnv = await TestEnv.create()

  await testEnv.run('install')
})

test('should split imports', async () => {
  await testEnv.writeFile('format.ts', 'import { a, b } from "./c"\nimport { d } from "./e"')

  await testEnv.run('format')
  await testEnv.run('format')

  assert.equal(
    await testEnv.readFile('format.ts'),
    `import { a } from './c'\nimport { b } from './c'\nimport { d } from './e'\n`
  )
})

test('should order imports', async () => {
  await testEnv.writeFile(
    'format.ts',
    'import { a } from "./c"\nimport { b } from "@scope/name"\nimport type { c } from "./d"'
  )

  await testEnv.run('format')
  await testEnv.run('format')

  assert.equal(
    await testEnv.readFile('format.ts'),
    `import type { c } from './d'\n\nimport { b }      from '@scope/name'\n\nimport { a }      from './c'\n`
  )
})

test('should align imports', async () => {
  await testEnv.writeFile(
    'format.ts',
    'import { first } from "./a"\nimport { second } from "./a"\nimport type { type } from "./a"\nimport third from "./a"'
  )

  await testEnv.run('format')
  await testEnv.run('format')

  assert.equal(
    await testEnv.readFile('format.ts'),
    `import type { type } from './a'\n\nimport { first }     from './a'\nimport { second }    from './a'\nimport third         from './a'\n`
  )
})
