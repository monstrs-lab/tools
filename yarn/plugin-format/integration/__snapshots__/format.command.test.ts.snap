// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`yarn commands format it should align imports 1`] = `
"import type { type } from './a'

import { first }     from './a'
import { second }    from './a'
import third         from './a'
"
`;

exports[`yarn commands format it should order imports 1`] = `
"import type { c } from './d'

import { b }      from '@scope/name'

import { a }      from './c'
"
`;

exports[`yarn commands format it should split imports 1`] = `
"import { a }    from './c'
import { b } from './c'
import { d }    from './e'
"
`;
